src/pyscipopt/expr.pxi

```diff
@@ -247,3 +343,324 @@ def quicksum(termlist):
+    #        self.constant += right.number
+    #    else:
+    #        self.coefs.append(1.0)
+    #        self.children.append(right)
+    #    return self
+
+    def __mul__(self, other):
+        left = buildGenExprObj(self)
+        right = buildGenExprObj(other)
+        ans = ProdExpr()
+
+        # multiply left factor
+        if left.getOp() == Operator.prod:
+            ans.children.extend(left.children)
+            ans.constant *= left.constant
+        elif left.getOp() == Operator.const:
+            ans.constant *= left.number
+        else:
+            ans.children.append(left)
+
+        # multiply right factor
+        if right.getOp() == Operator.prod:
+            ans.children.extend(right.children)
+            ans.constant *= right.constant
+        elif right.getOp() == Operator.const:
+            ans.constant *= right.number
+        else:
+            ans.children.append(right)
+
+        return ans
+
+    #def __imul__(self, other):
+    #''' in-place multiplication, i.e., expr *= other '''
+    #    assert isinstance(self, Expr)
+    #    right = buildGenExprObj(other)
+    #    # transform self into prod
+    #    if self.getOp() != Operator.prod:
+    #        newprod = ProdExpr()
+    #        if self.getOp() == Operator.const:
+    #            newprod.constant *= self.number
+    #        else:
+    #            newprod.children.append(self.copy()) # TODO: what is copy?
+    #        self = newprod
+    #    # multiply right factor
+    #    if right.getOp() == Operator.prod:
+    #        self.children.extend(right.children)
+    #        self.constant *= right.constant
+    #    elif right.getOp() == Operator.const:
+    #        self.constant *= right.number
+    #    else:
+    #        self.children.append(right)
+    #    return self
+
+    def __pow__(self, other, modulo):
+        expo = buildGenExprObj(other)
+        if expo.getOp() != Operator.const:
+            raise NotImplementedError("exponents must be numbers")
+        if self.getOp() == Operator.const:
+            return Constant(self.number**expo.number)
+        ans = PowExpr()
+        ans.children.append(self)
+        ans.expo = expo.number
+
+        return ans
+
+    #TODO: ipow, idiv, true_div, etc
+    def __div__(self, other):
+        divisor = buildGenExprObj(other)
+        # we can't divide by 0
+        if divisor.getOp() == Operator.const:
+            assert divisor.number != 0.0
+
+        return self * divisor**(-1)
+
+    def __neg__(self):
+        return -1.0 * self
+
+    def __sub__(self, other):
+        return self + (-other)
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __rsub__(self, other):
+        return -1.0 * self + other
+
+    def __richcmp__(self, other, op):
+        '''turn it into a constraint'''
+        return _expr_richcmp(self, other, op)
+
+    def degree(self):
```

Comment:
why even add this method?

Reply:
just to keep it compatible with the implementation of addCons in scip.pxd or whathever the termination is